src/Kernel/Object.class.st

```diff
@@ -1578,6 +1578,18 @@ Object >> okToChange [
 	^ true
 ]
 
+{ #category : #debugging }
+Object >> openDebugger [
```

Comment:
Actually, I'm not really sure about this. The kernel should be independent of things like debugger as much as possible. The system raises an exception and it's the responsibility of tools what they will do. The opening of the debugger is only one option, in a production application, it may be something else. Then this message could be misleading.

Reply:
The reasoning behind this method is that some code in the image, for example DelayBasicScheduler>>runBackendLoopAtTimingPriority, DelayBasicScheduler>>schedule: contain code like
```"When two debuggers appear, step/proceed through this higher priority one first."
debug ifTrue: [ self halt ].```
This looks like some debugging code left in the image on purpose as a framework to ease debugging later. So a proxy method to call halt (which is the only thing Object>>openDebugger does) allows this method to specify that this "halt" is intended to stay in production.

Reply:
I did that with @MarcusDenker during a sprint, so maybe he can contribute to the discussion.

Reply:
You should really discuss with Guille. But as a KEY GENERAL principle we do not want to have dependencies to tools from the Kernel. This is an insane idea! 
We spent days removing them. Can't we think in terms of layers?